CONTRIBUTING.md

```diff
@@ -0,0 +1,187 @@
+# Contributing
+
+Thanks for taking a look at StatsBase!
+
+In general this guide summarizes and extends the [Julia Contributing Guidelines](https://github.com/JuliaLang/julia/blob/master/CONTRIBUTING.md#general-formatting-guidelines-for-julia-code-contributions) and [Julia Style Guide](https://docs.julialang.org/en/latest/manual/style-guide.html).
+
+## Reporting Issues
+
+* It's always helpful to start by reviewing existing issues is and if necessary referencing back to previous discussions.
+* Including minimal sample code (either to reproduce a bug or demonstrating desired behaviour) is greatly appreciated.
+* For bugs, it's helpful to checkout the latest development version (`Pkg.checkout("StatsBase")`) and post the output from `versioninfo()`.
+
+## Pull Requests (PR)s
+
+* Try to keep changes in PRs constrained if possible. This will reduce the amount of work for the reviewer and likely speed up the review process.
+* Using `git diff upstream/master` and `git diff --stat upstream/master` (assuming you've run `git remote add upstream https://github.com/JuliaStats/StatsBase.jl`) can be useful commands for comparing your branch against the upstream master before pushing.
+* Running tests locally with `Pkg.test("StatsBase")` before committing can be a helpful way of reducing noise.
+* [Coverage](https://github.com/JuliaCI/Coverage.jl) can be useful for checking if you've lowered the overall test coverage against [master](https://coveralls.io/r/JuliaStats/StatsBase.jl?branch=master)
+
+## Style
+
+### Summary
+
+* Use 4 spaces per indentation level, no tabs.
+* Try to adhere to the 92 character line length limit.
+* Use upper camel case convention for [modules](http://julia.readthedocs.org/en/latest/manual/modules/) and [types](http://julia.readthedocs.org/en/latest/manual/types/).
+* Use lower case for method names, preferably without underscores.
+* Comments are a good way to explain the intentions of your code and can aid in Pull Request (PR) reviews.
+* No whitespace at the end of a line (trailing whitespace).
+* Avoid padding brackets with spaces. ex. `Int64(value)` preferred over `Int64( value )`.
+* Include spaces after commas and around operators
+
+### Methods Definitions
+
+Short form method definitions when the method body only contains 1 statement:
+
+```julia
+# Yes
+Base.stdm(v::RealArray, w::AbstractWeights, m::Real; corrected::DepBool=nothing) =
+    sqrt(varm(v, w, m, corrected=depcheck(:stdm, corrected)))
```

Comment:
Can you hide the `depcheck` hack? It's meant to be removed anyway, and the person who will remove it will probably forget to sync the style guide.

Same below.